docs/concepts/realms.md

```diff
@@ -3,47 +3,196 @@ id: realms
 ---
 
 # Realms
+In Gno.land, realms are entities that are addressable and identifiable by a 
+[Gno address](../reference/standard-library/std/address.md). These can be user 
+realms (EOAs), as well as smart contract realms. Realms have several 
```

Comment:
I've based this PR on the [clarification](https://github.com/gnolang/gno/issues/1579#issuecomment-1912710562) @moul gave where we discussed the reasons why we use the term `user realms`, and why the `Realm` type in Gno has a `IsUser` check. From my understanding, realms are not simply "smart contracts" - they are entities that are addressable. This would include the accounts generated by `gnokey`, Adena, etc. These would simply be realms that do not contain code. 

Maybe using the term EOA was a mistake. It seems that `user account` or `user realm` would be more appropriate.

@moul, can you please provide clarity on this issue? 🙏